db/consistency_level.cc

```diff
@@ -115,22 +115,15 @@ bool is_datacenter_local(consistency_level l) {
 }
 
 template <typename Range, typename PendingRange = std::array<gms::inet_address, 0>>
-std::unordered_map<sstring, dc_node_count> count_per_dc_endpoints(
+std::unordered_map<const locator::datacenter*, dc_node_count> count_per_dc_endpoints(
         const locator::effective_replication_map& erm,
         const Range& live_endpoints,
         const PendingRange& pending_endpoints = std::array<gms::inet_address, 0>()) {
     using namespace locator;
 
-    auto& rs = erm.get_replication_strategy();
     const auto& topo = erm.get_topology();
 
-    const network_topology_strategy* nrs =
-            static_cast<const network_topology_strategy*>(&rs);
-
-    std::unordered_map<sstring, dc_node_count> dc_endpoints;
-    for (auto& dc : nrs->get_datacenters()) {
-        dc_endpoints.emplace(dc, dc_node_count());
-    }
```

Comment:
This changes the logic, before we would return an entry for each DC even if it didn't have any live endpoints, and now we don't. This may result in assure_sufficient_live_nodes_each_quorum() passing instead of failing.